catalog/resource_catalog.go

```diff
@@ -17,56 +18,16 @@ func ucDirectoryPathSuppressDiff(k, old, new string, d *schema.ResourceData) boo
 	return false
 }
 
-type CatalogsAPI struct {
-	client  *common.DatabricksClient
-	context context.Context
-}
-
-func NewCatalogsAPI(ctx context.Context, m any) CatalogsAPI {
-	return CatalogsAPI{m.(*common.DatabricksClient), context.WithValue(ctx, common.Api, common.API_2_1)}
-}
-
 type CatalogInfo struct {
```

Comment:
@nkvuong did you try using UpdateCatalog for the schema this way we won't have to worry about the new fields?...